File: backend/src/main/java/be/technobel/parsemaster/service/implementation/UserDetailsServiceImpl.java
package be.technobel.parsemaster.service.implementation;

import be.technobel.parsemaster.exception.Exceptions;
import be.technobel.parsemaster.repository.UserRepository;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.stereotype.Service;

@Service
public class UserDetailsServiceImpl implements UserDetailsService {
  private final UserRepository userRepository;

  public UserDetailsServiceImpl(UserRepository userRepository) {
    this.userRepository = userRepository;
  }

  @Override
  public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
    return userRepository.findByEmailOrPseudonym(username)
      .orElseThrow(Exceptions.USERNAME_NOT_FOUND::create);
  }
}

File: src/main/kotlin/be/technobel/playzone/bll/services/impl/UserDetailsServiceImpl.kt
package be.technobel.playzone.bll.services.impl

import be.technobel.playzone.dal.repositories.UserRepository
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.security.core.userdetails.UsernameNotFoundException
import org.springframework.stereotype.Service

@Service
class UserDetailsServiceImpl(private val userRepository: UserRepository) : UserDetailsService {
	override fun loadUserByUsername(username: String): UserDetails =
		userRepository.findByEmail(username)
		?: throw UsernameNotFoundException("User not found")
}